Makefile

```diff
@@ -148,6 +148,16 @@ distclean:
 install: checkprefix nvim
 	+$(BUILD_TOOL) -C build install
 
+clang-tidy: generate-compilation-db
+	clang-tidy --config-file .clang-tidy -p build $(FILE)
+
+clang-tidy-full: generate-compilation-db
+	+$(BUILD_CMD) -C build clang-tidy-full
+
+generate-compilation-db: build/.ran-cmake
+	ninja -C build -t compdb C_COMPILER__nvim_$(CMAKE_BUILD_TYPE) > build/compile_commands.json
```

Comment:
Is this needed anymore? Doesn't cmake automatically create a compilation database now?

Reply:
this is specifically for filtering that database, see https://github.com/neovim/neovim/issues/10632#issuecomment-877663416

Reply:
Wait, that's a completely separate issue. This PR is about introducing clang-tidy, that issue is about "massing" the compilation database for whatever reason.

Reply:
> for whatever reason

it's specifically for clang gang, see its title :sweat_smile: 
- https://github.com/neovim/neovim/issues/10632